src/extensionProvider.ts

```diff
@@ -161,8 +166,13 @@ export class ExtensionProvider {
     const signatureHex = extensionResponse.signature;
     const signature = Buffer.from(signatureHex, "hex");
 
-    message.applySignature(signature);
-    return message;
+    return new Message({
+      data: Buffer.from(messageToSign.data),
+      address: messageToSign.address ?? Address.fromBech32(this.account.address),
+      signer: "extension",
```

Comment:
Is there a more precise name? E.g. `MultiversX DeFi Wallet`?

Reply:
Or maybe this provider is used for other extensions, as well.

Reply:
Could be used by anybody